crypto/zk/circuit/config.go

```diff
@@ -47,6 +47,18 @@ func (config ZkCircuitConfig) KeySize() int {
 	return config.Levels / 8
 }
 
+// SupportsCensusSize returns if the provided censusSize is supported by the
+// current circuit configuration. It ensures that the provided value is lower
+// than 2^config.Levels.
+func (config ZkCircuitConfig) SupportsCensusSize(maxCensusSize uint64) bool {
+	return true
```

Comment:
Wow, I don't really understand the `true` :1st_place_medal: 

I think we should make this (and everything related to the number of levels of the tree), dynamic.

So in the ZkCircuitConfig constructor I'd pre-compute the number and store it `new(big.Int).Exp(big.NewInt(2), big.NewInt(int64(config.Levels)), nil)` then SupportsCensusSize just performs a bigint comparison: `return config.circuitMaxCensusSize.Cmp(new(big.Int).SetUint64(maxCensusSize)) != 1`

Reply:
I think it is solved now!